### PR TITLE
update envoy to 1.18.3

### PR DESCRIPTION
**Pull request by davinci26**
Signed-off-by: Sotiris Nanopoulos <sonanopo@microsoft.com>

**Description**:

See #3343

Updates OSM to user Envoy version 1.18.3.

I will do a follow up PR to change the generated config to use some newer features.

I validated through `./demo/run-osm-demo.sh` and `./bin/osm proxy get config_dump bookbuyer-85d5c5f4f4-l7bgs -n bookbuyer > envoy-new.json`

```
    "node": {
     "hidden_envoy_deprecated_build_version": "98c1c9e9a40804b93b074badad1cdf284b47d58b/1.18.3/Clean/RELEASE/BoringSSL",
     "user_agent_name": "envoy",
     "user_agent_build_version": {
      "version": {
       "major_number": 1,
       "minor_number": 18,
       "patch": 3
      },
      "metadata": {
       "revision.sha": "98c1c9e9a40804b93b074badad1cdf284b47d58b",
       "build.type": "RELEASE",
       "revision.status": "Clean",
       "ssl.version": "BoringSSL"
      }
 ```

**Affected area**:

- New Functionality      [ ]
- Documentation          []
- Install                []
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Demo                   [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

### CODE REVIEW

**Comment by shashankram** [comment]
I am not against this change, but what's the motivation to update to 1.18 from 1.17? Is 1.18 stable enough?

File: cmd/init-osm-controller/init-osm-controller.go [hunk 1]
```diff
@@ -64,7 +64,30 @@ func createDefaultMeshConfig(presetMeshConfig *v1alpha1.MeshConfig) *v1alpha1.Me
 		ObjectMeta: metav1.ObjectMeta{
 			Name: meshConfigName,
 		},
-		Spec: presetMeshConfig.Spec,
+		Spec: v1alpha1.MeshConfigSpec{
```
Comment (davinci26) [inline cmd/init-osm-controller/init-osm-controller.go:67]: This change should be reverted, this due to a bad merge conflict

**Comment by davinci26** [comment]
> I am not against this change, but what's the motivation to update to 1.18 from 1.17? Is 1.18 stable enough?

It is actually need to port on Windows so I made this change to start familiarizing myself with the codebase.

One of the issues that we will encounter is the way we use the `FileAccessLog` to redirect the logs to `/dev/stdout` which is not cross-platform.

For that reason we have developed [stdoutaccesslog](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/access_loggers/stream/v3/stream.proto.html?highlight=stdoutaccesslog) which uses the platform specific stdout handler.
 
I was planning to do a follow up to update that but for that we need version 1.18.3

**Comment by shashankram** [comment]
> > I am not against this change, but what's the motivation to update to 1.18 from 1.17? Is 1.18 stable enough?
> 
> It is actually need to port on Windows so I made this change to start familiarizing myself with the codebase.
> 
> One of the issues that we will encounter is the way we use the `FileAccessLog` to redirect the logs to `/dev/stdout` which is not cross-platform.
> 
> For that reason we have developed [stdoutaccesslog](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/access_loggers/stream/v3/stream.proto.html?highlight=stdoutaccesslog) which uses the platform specific stdout handler.
> 
> I was planning to do a follow up to update that but for that we need version 1.18.3

I'd suggest only updating to the latest version of Envoy if we need something that is not in 1.17 already. 
If there are Windows-specific requirements only available in 1.18, I'd suggest creating a Github issue to start with and linking this PR to that issue so we know why this specific change is required.

Hope that makes sense.

**Comment by davinci26** [comment]
See https://github.com/openservicemesh/osm/issues/3343

**Comment by davinci26** [comment]
I am still trying to understand the differences between the CI and my local dev env. I will ping you when this is ready for a review

**Review by draychev — APPROVED** [review]

**Comment by draychev** [comment]
@davinci26 I love the fact that you validated that the config is the same!! Thank you!

We are going to do extensive testing as soon as next week w/ v0.9.0 of OSM - so we'll find out quickly if there are issues w / Envoy v0.18

**Review by shashankram — APPROVED** [review]

**Comment by draychev** [comment]
fix #3343 3343

**Pull request merged**